subprojects/core/src/main/java/org/gradle/process/internal/DefaultExecHandle.java

```diff
@@ -178,11 +178,11 @@ public ExecHandleState getState() {
     private void setState(ExecHandleState state) {
         lock.lock();
         try {
-            LOGGER.debug("Changing state to: {}", state);
             this.state = state;
             this.stateChanged.signalAll();
         } finally {
             lock.unlock();
+            LOGGER.debug("Changing state to: {}", state);
```

Comment:
I would move the logging to before we aquire the lock. If we leave it afterwards, then I would change the message to `Changed state to: `.

Reply:
What really matters is `LOGGER.debug("Changing state to: {}", state);` must be after `this.stateChanged.signalAll();`, not before. But yes, it should be `Changed state to`.